pilot/pkg/bootstrap/server.go

```diff
@@ -455,13 +451,14 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 			if err != nil {
 				return err
 			}
-			credentials := creds.CreateForClient(u.Hostname(), watcher)
+			credentials := creds.CreateForClient(configSource.TlsSettings.Sni, watcher)
 			securityOption = grpc.WithTransportCredentials(credentials)
 		}
+
 		msgSizeOption := grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(args.MCPMaxMessageSize))
-		conn, err := grpc.DialContext(ctx, u.Host, securityOption, msgSizeOption)
+		conn, err := grpc.DialContext(ctx, configSource.Address, securityOption, msgSizeOption)
```

Comment:
Is there an option to validate SAN in go grpc?

Reply:
AFAK, there is no such method, unless the SAN is in the cert.